DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.WebAPI/EventParsers/BaseEventParser.cs

```diff
@@ -0,0 +1,33 @@
+﻿using Newtonsoft.Json.Linq;
+
+namespace GitHubExtension.Activity.External.WebAPI.EventParsers
+{
+    public abstract class BaseEventParser
+    {
+        public abstract JToken ParseEvent(JToken eventToken);
+
+        protected JToken InitialiseEventData(JToken eventToken)
```

Comment:
So I will conditionally get the type of event and then will serialize it to desired strong type, am I right?